mage/k3d/lib.go

```diff
@@ -103,9 +104,28 @@ func reloadLocalImage(image string) {
 }
 
 func setupKubeconfig() {
-	config := shutil.OutputPanic("k3d", "get-kubeconfig")
-	os.Setenv("KUBECONFIG", config)
-	fmt.Printf("To set up kubectl in your shell to use this cluster, set:\n\tKUBECONFIG=%s\n", config)
+	k3dConfig := shutil.OutputPanic("k3d", "get-kubeconfig")
+	currentConfig := kubectl.GetKubeconfig()
+
+	// merge current config + new k3d config
+	// the new config must appear before the current config to update it
+	// when merging
+	os.Setenv("KUBECONFIG", fmt.Sprintf("%s:%s", k3dConfig, currentConfig))
+	merged := shutil.OutputPanic("kubectl", "config", "view", "--flatten")
```

Comment:
I don't quite follow this. This line suggests that `KUBECONFIG` can have multiple values, so why do we need to merge the kube config files at all?

Reply:
So the goal of this is to set up the user's kubeconfig so that they can immediately start using kubectl to talk to the new cluster that we just stood up for them. We can easily have kubeconfig set up inside our go process by doing the normal os.Setenv call, but as soon as the go process finished running, the user's shell would not be set up to let them start using kubectl without running at least 1 manual command to set up kubeconfig.

By doing it this way, we are leveraging kubectl's ability to merge multiple configs and then we update the user's current config on disk with the merged value, and then finally we tell kubectl to switch to the new context. 

KIND and gcloud already did this automatically when you create a new cluster, but k3d was more conservative and created a separate config file and didn't do any merging.